src/libcore/iter.rs

```diff
@@ -641,6 +635,13 @@ pub trait IteratorExt: Iterator + Sized {
     /// Returns the first element satisfying the specified predicate.
     ///
     /// Does not consume the iterator past the first found element.
+    ///
+    /// # Example
```

Comment:
I think stylistically we tend to have a newline after a heading, e.g.

````
# example

```rust
...
````